### ISSUE TITLE
batch insert gives  error

### ISSUE THREAD

**Issue by rabiSharelookapp** [jeremydaly/data-api-client#86]
while query is done for batchInsert  it insert the multiple row but throws an error "BadRequestException:Batch entry 1 <unknown> was aborted: Too many update results were returned. Call getNextException to see othe errors in the batch". though the items passed as same structure as in  document 
![Screenshot from 2021-04-01 11-16-36](https://user-images.githubusercontent.com/79686947/113249659-c9be0280-92de-11eb-8d2c-e98bd7949f1b.png)

**Comment by jeremydaly** [comment]
This looks like a Postgres error. Are you able to run a smaller batch?

**Comment by rabiSharelookapp** [comment]
I could run a small batch with 3 or might be more but it does insertion on
table but throws that error even batch with length 1 .

**Comment by rabiSharelookapp** [comment]
I could run a small batch with 3 or might be more but it does insertion on table but throws that error even batch with length 1 .

**Comment by rabiSharelookapp** [comment]
what would be the solution for this @jeremydaly

**Comment by jeremydaly** [comment]
@ffxsam Any thoughts on this? Haven't seen this error with batches before.

**Comment by rabiSharelookapp** [comment]
@jeremydaly today I got another situation that if table not exist then batch insertion execution is only done for 1 batch more than one it gives error but after 1 batch entry or table exist then more than 1 batch be inserted.Is this from my side or  the issue                      (I have used create table if not exist query to create table )

**Comment by ffxsam** [comment]
No, sorry I can't be of any help. I actually don't use batch inserts very often. Plus, I'm using knex to build all my queries before passing the resulting string to `dataApi.query()`.

**Comment by rabiSharelookapp** [comment]
@Sam , could you suggest a library for knex that is compatible with
data-api. I tried to do with knex.js only but couldn't connect using aws
arn

**Comment by rabiSharelookapp** [comment]
And also use case  here is that need to perform migration , could you
please suggest how to I perform that trough data-api-client

**Comment by ffxsam** [comment]
@rabiSharelookapp knex works fine with dataApi. You can use it like this:

```js
const knex = require('knex')({ client: 'pg' });

async function doTheThing() {
  const results = await dataApi.query(knex('my_table').select('a', 'b', 'c'));
}
```

As for your other question, that's a bit of a general and very involved question that I unfortunately don't have time to help with.

**Comment by rabiSharelookapp** [comment]
okay thanks
